android/src/main/new_arch/react/renderer/components/RNLiveMarkdownSpec/MarkdownTextInputDecoratorShadowNode.cpp

```diff
@@ -0,0 +1,125 @@
+
+  const auto &fabricUIManager =
+      this->getContextContainer()->at<JFabricUIManager::javaobject>("FabricUIManager");
+
+  const auto customUIManager = jni::make_global(createMethod(
+      customFabricUIManagerClass, fabricUIManager,
+      decoratorPropsRM.get(), parserId));
+  const auto contextContainer =
+      std::make_shared<ContextContainer const>();
+  contextContainer->insert("FabricUIManager", customUIManager);
+
+  customContextContainer_ = contextContainer;
+  previousMarkdownStyle_ = markdownStyle;
+  previousParserId_ = parserId;
+}
+
+void MarkdownTextInputDecoratorShadowNode::updateCustomContextContainerIfNeeded() {
+  const auto markdownStyle =
+      this->getProps()->rawProps["markdownStyle"];
+  const auto parserId =
+      this->getProps()->rawProps["parserId"].asInt();
+
+  if (parserId != previousParserId_ || markdownStyle != previousMarkdownStyle_) {
+    createCustomContextContainer();
+  }
+}
+
+void MarkdownTextInputDecoratorShadowNode::overwriteTextLayoutManager() {
+  const auto &children = getChildren();
+  if (children.empty()) {
+    return;
+  }
+  react_native_assert(
+      children.size() == 1 &&
+      "MarkdownTextInputDecoratorView received more than one child");
+
+  const auto child = std::dynamic_pointer_cast<const AndroidTextInputShadowNode>(
+      children.at(0));
+
+  react_native_assert(
+      child != nullptr &&
+      "MarkdownTextInputDecoratorView received child other than a TextInput");
+  child->ensureUnsealed();
+
+  const auto mutableChild = std::const_pointer_cast<AndroidTextInputShadowNode>(child);
+  mutableChild->setTextLayoutManager(
+      std::make_shared<TextLayoutManager>(customContextContainer_));
+}
+
+void MarkdownTextInputDecoratorShadowNode::appendChild(
+    const ShadowNode::Shared &child) {
+  YogaLayoutableShadowNode::appendChild(child);
+
+  overwriteTextLayoutManager();
+}
+
+void MarkdownTextInputDecoratorShadowNode::replaceChild(
+    const ShadowNode &oldChild, const ShadowNode::Shared &newChild,
+    size_t suggestedIndex) {
+  YogaLayoutableShadowNode::replaceChild(oldChild, newChild, suggestedIndex);
+
+  overwriteTextLayoutManager();
+}
+
+void MarkdownTextInputDecoratorShadowNode::layout(LayoutContext layoutContext) {
+  YogaLayoutableShadowNode::layout(layoutContext);
+
+  const auto &children = getChildren();
+  react_native_assert(
+      children.size() == 1 &&
+      "MarkdownTextInputDecoratorView didn't receive exactly one child");
+
+  auto child = std::static_pointer_cast<const YogaLayoutableShadowNode>(getChildren()[0]);
+  child->ensureUnsealed();
+  auto mutableChild = std::const_pointer_cast<YogaLayoutableShadowNode>(child);
+
+  // TODO: this may not be the correct way to do this
+  auto childMetrics = child->getLayoutMetrics();
+  setLayoutMetrics(childMetrics); // makes a copy
+
+  childMetrics.frame.origin = Point{}; // origin is relative to the parent
+  mutableChild->setLayoutMetrics(childMetrics);
```

Comment:
Let's add some commentary here as well

Reply:
Is that ok? [7250899](https://github.com/Expensify/react-native-live-markdown/pull/566/commits/7250899ab32fa4aa7481befeaa20e5be46cc8388)